platform/windows/export/template_modifier.cpp

```diff
@@ -0,0 +1,674 @@
+	String product_name = p_preset->get("application/product_name");
+	String file_description = p_preset->get("application/file_description");
+	String copyright = p_preset->get("application/copyright");
+	String trademarks = p_preset->get("application/trademarks");
+
+	HashMap<String, String> strings;
+	if (!file_version.is_empty()) {
+		strings["FileVersion"] = file_version;
+	}
+	if (!product_version.is_empty()) {
+		strings["ProductVersion"] = product_version;
+	}
+	if (!company_name.is_empty()) {
+		strings["CompanyName"] = company_name;
+	}
+	if (!product_name.is_empty()) {
+		strings["ProductName"] = product_name;
+	}
+	if (!file_description.is_empty()) {
+		strings["FileDescription"] = file_description;
+	}
+	if (!copyright.is_empty()) {
+		strings["LegalCopyright"] = copyright;
+	}
+	if (!trademarks.is_empty()) {
+		strings["LegalTrademarks"] = trademarks;
+	}
+
+	return strings;
+}
+
+Error TemplateModifier::_modify_template(const Ref<EditorExportPreset> &p_preset, const String &p_template_path, const String &p_icon_path) const {
+	Vector<SectionEntry> section_entries = _get_section_entries(p_template_path);
+	ERR_FAIL_COND_V(section_entries.size() < 2, ERR_CANT_OPEN);
+	ERR_FAIL_COND_V(section_entries[section_entries.size() - 2].name != String(".rsrc"), ERR_CANT_OPEN);
+	ERR_FAIL_COND_V(section_entries[section_entries.size() - 1].name != String(".reloc"), ERR_CANT_OPEN);
+	// TODO fail on not sorted by physical address?
+
+	GroupIcon group_icon = _create_group_icon(p_icon_path);
+
+	VersionInfo version_info = _create_version_info(_get_strings(p_preset));
+
+	SectionEntry resources_section_entry = section_entries.get(section_entries.size() - 2);
+
+	Vector<uint8_t> resources = _create_resources(resources_section_entry.virtual_address, group_icon, version_info);
+
+	resources_section_entry.virtual_size = resources.size();
+	resources.resize_zeroed(_snap(resources.size(), BLOCK_SIZE));
+	resources_section_entry.size_of_raw_data = resources.size();
+
+	SectionEntry relocations_section_entry = section_entries.get(section_entries.size() - 1);
+	relocations_section_entry.pointer_to_raw_data = resources_section_entry.pointer_to_raw_data + resources_section_entry.size_of_raw_data;
+	relocations_section_entry.virtual_address = resources_section_entry.virtual_address + _snap(resources_section_entry.virtual_size, PE_PAGE_SIZE);
+
+	uint32_t size_of_image = relocations_section_entry.virtual_address + _snap(relocations_section_entry.virtual_size, PE_PAGE_SIZE);
+
+	uint32_t pe_header_offset = _get_pe_header_offset(p_template_path);
+
+	Error error;
+	Ref<FileAccess> template_file = FileAccess::open(p_template_path, FileAccess::READ_WRITE, &error);
+	ERR_FAIL_COND_V(error != OK, ERR_CANT_OPEN);
+
+	Vector<uint8_t> relocations;
+	relocations.resize(relocations_section_entry.size_of_raw_data);
+	template_file->seek(relocations_section_entry.pointer_to_raw_data);
+	template_file->store_buffer(relocations);
```

Comment:
What's this supposed to do? I assume it's intended to read the existing `.reloc` section to write it back later, but it's using `pointer_to_raw_data` after it was modified, and writing garbage (since non-zeroed resize is used) to the file instead of reading.